lib/src/source_visitor.dart

```diff
@@ -1836,11 +1911,16 @@ class SourceVisitor extends ThrowingAstVisitor {
       modifier(node.keyword);
 
       visit(node.type);
+      bool hasType = node.type != null;
+      if (_insideNewTypedefFix && !hasType) {
+        _writeText("dynamic", node.identifier.offset);
```

Comment:
This needs a test for what happens to a comment before the identifier.

Reply:
Ick, good point.
Should it stay with the name, or should it get put before the type?
It's like metadata, so it could precede the entire typed parameter, but you *can* write `typedef F(foo(int /*c*/ bar));`. That's definitely where the current code puts it. 

Do you want to move the comment before the `dynamic` in this case?